src/yunohost/certificate.py

```diff
@@ -639,6 +639,16 @@ def _prepare_certificate_signing_request(domain, key_file, output_folder):
     # Set the domain
     csr.get_subject().CN = domain
 
+    from yunohost.domain import _get_maindomain
+    if domain == _get_maindomain():
+        # Include xmpp-upload subdomain in subject alternate names
+        subdomain="xmpp-upload." + domain
+        if _dns_ip_match_public_ip(get_public_ip(), subdomain):
+            logger.info("Subdmain {} is ready for ACME and will be included in the certificate.".format(subdomain))
```

Comment:
This string should be internationalized with `m18n` like a few lines below. But idk, personally I'd just vote to remove the message, but that's up to you.